src/renderers/dom/client/eventPlugins/ChangeEventPlugin.js

```diff
@@ -45,6 +47,35 @@ var eventTypes = {
   },
 };
 
+function getInstIfValueChanged(targetInst, target) {
+  var value = getInputValue(target);
+  var lastValue = target[LAST_VALUE_KEY];
+
+  if (!target.hasOwnProperty(LAST_VALUE_KEY) || value !== lastValue) {
+    target[LAST_VALUE_KEY] = value;
```

Comment:
> Missing an update should be fine since the next raised event will update the cached value

But it's not fine if you revert what the user typed and he types it again, it will then think the value hasn't changed, while uncommon that would be a severe bug.

I'm also skeptical of something like this if there are edge-cases where we are *still* triggering when the value hasn't actually changed (which can happen if the cache is wrong). If it's not 100% then the end benefit of this seems little more than a performance optimization, IMHO a good thing but perhaps not in relation to there being a fragile element to the implementation.

Also consider that we still don't know when auto-filling occurs in some browsers, this is a troublesome area.